CMakeLists.txt

```diff
@@ -6,24 +6,51 @@ set(CMAKE_MODULE_PATH "$ENV{CARMA_OPT_DIR}/cmake")
 set(CMAKE_CXX_STANDARD 17)
 
 
-project(carma-clock
+project(carma_clock
     DESCRIPTION "CARMA time library"
     HOMEPAGE_URL https://github.com/usdot-fhwa-stol/carma-time-lib
 	VERSION 0.0.1
     LANGUAGES CXX
 )
-find_package(Threads)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
 
+include(cmake/dependencies.cmake)
+add_library(${PROJECT_NAME} SHARED )
+
+target_include_directories(${PROJECT_NAME} 
+  PUBLIC
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+)
 
-add_library(${PROJECT_NAME} SHARED)
 set_target_properties(${PROJECT_NAME} PROPERTIES
     SOVERSION 1
 )
 
+if (DEFINED ENV{BUILD_ARCHITECTURE})
```

Comment:
Is this BUILD_ARCHITECTURE only defined with value arm? Can it is defined with value x64?